File: MediCareWebSpa/src/app/DTOs/request/role-permission-request.dto.ts
import { RoleType } from "../../enums/role-type";

export class RolePermissionRequest {
  roleType: RoleType;
  permissionId: number;

  constructor(
    roleType: RoleType,
    permissionId: number,
  ) {
    this.roleType = roleType;
    this.permissionId = permissionId;
  }
}

File: MediCareWebApi/DTOs/Request/RolePermissionRequestDTO.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.DTOs.Request
{
	public class RolePermissionRequestDTO
	{
		[Required(ErrorMessage = "Typ roli jest wymagany.")]
		public RoleType RoleType { get; set; }

		[Required(ErrorMessage = "Id uprawnienia jest wymagane.")]
		public int PermissionId { get; set; }
	}
}
